bloggies_backend/src/routes/stripe.ts

```diff
@@ -82,34 +89,45 @@ stripeRouter.post("/create-customer", ensureLoggedIn, async function (req: Reque
 /** POST create a subscription for a customer and save payment method  */
 stripeRouter.post("/create-subscription", ensureLoggedIn, async function (req: Request, res: Response, next: NextFunction) {
   const { paymentMethodId, customerId } = req.body;
-  try {
-    // save payment method info for a customer
-    await stripe.paymentMethods.attach(paymentMethodId, {
-      customer: customerId
+  const { user_id } = req.user;
+
+  const userStatusRes = await User.checkMembershipStatus(user_id);
+  if (ableToStartSub(userStatusRes.membership_status)) {
+    try {
+      // save payment method info for a customer
+      await stripe.paymentMethods.attach(paymentMethodId, {
+        customer: customerId
+      });
+    } catch (err) {
+      const expressErr = new ExpressError(err.message, err.statusCode);
+      return next(expressErr);
+    }
+    // update the customer with the payment method
+    await stripe.customers.update(customerId, {
```

Comment:
If we are going to await this, it should probably go in a try/catch!

Reply:
Right!! I'll go add the try-catch there